File: App/src/app/core/model/order-detail.ts
import { Attribute } from "./attribute";
import { Review } from "./review";

export interface OrderDetail {
    id?: number;
    orderId?: number;
    productId: number;
    productName: string;
    productImage: string;
    productPrice: number;
    productDiscountPrice: number;
    qty: number;
    attribute?: string;

    reviews?: Review[];

    productAlias?: string;
    attributes: Attribute[];
    isReview?: boolean;
}

File: Service/Model/OrderDetail.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Service.Model
{
    [Table("OrderDetail")]
    public class OrderDetail
    {
        public int Id { get; set; }
        public int OrderId { get; set; }
        public int ProductId { get; set; }
        public string ProductName { get; set; } = "";
        public string ProductImage { get; set; } = "";
        public double? ProductPrice { get; set; }
        public double? ProductDiscountPrice { get; set; }
        public int? Qty { get; set; }
        public string Attribute { get; set; } = "";

        public virtual Order? Order { get; set; }
        public virtual Product? Product { get; set; }
        public virtual ICollection<Review>? Reviews { get; set; }
    }
}
